contributions/whiten.py

```diff
@@ -0,0 +1,191 @@
+    parser.add_argument('-d', '--max-filter-duration', type=float, default=0.25, help="Maximum length of the time-domain filter in seconds. Default: 0.25.")
+    parser.add_argument('--hard-trunc', action='store_true', help="Use a hard truncation of the time-domain filter. Otherwise, a Hann window is used.")
+    parser.add_argument('--keep-corrupted', action='store_true', help="Keep the region of the time series corrupted by the whitening. Otherwise, it is excised.")
+    parser.add_argument('-f', '--low-frequency-cutoff', type=float, default=20., help="Low frequency cutoff to be passed to the inverse spectrum truncation. This should be matched to a known low frequency cutoff of the data if there is one. Default: 20.")
+    parser.add_argument('--dtype', type=int, default=None, help="Data type specified by number of bits. If supplied, the input data will be cast to the corresponding floating point data type before whitening.")
+
+    args = parser.parse_args()
+
+    ### Set up logging
+    if args.debug:
+        log_level = logging.DEBUG
+    elif args.verbose:
+        log_level = logging.INFO
+    else:
+        log_level = logging.WARN
+    logging.basicConfig(format='%(levelname)s | %(asctime)s: %(message)s', level=log_level, datefmt='%d-%m-%Y %H:%M:%S')
+
+    ### Check existence of output file
+    if os.path.isfile(args.outputfile):
+        raise RuntimeError("Output file exists.")
+    else:
+        pass
+
+    if args.hard_trunc:
+        trunc_method = None
+    else:
+        trunc_method = 'hann'
+
+    logging.debug(("Initializing caster to data type %s" % str(args.dtype)))
+    caster = SafeCaster(args.dtype)
+
+    with h5py.File(args.inputfile, 'r') as infile, h5py.File(args.outputfile, 'x') as outfile:
+        copy_attrs(infile, outfile)
+        iterable1 = infile.items()
+        if args.verbose:
+            iterable1 = tqdm(iterable1, desc='Iterating over detectors')
+        for detector_group_name, in_detector_group in iterable1:
+            out_detector_group = outfile.create_group(detector_group_name)
+            copy_attrs(in_detector_group, out_detector_group)
+            iterable2 = in_detector_group.items()
+            if args.verbose:
+                iterable2 = tqdm(iterable2, desc='Iterating over segments', leave=False)
+            for segment_name, in_segment in iterable2:
+                segment_colored_data = in_segment[()]
+                segment_white_data = whiten(segment_colored_data, delta_t=1./infile.attrs['sample_rate'], segment_duration=args.segment_duration, max_filter_duration=args.max_filter_duration, trunc_method=trunc_method, remove_corrupted=(not args.keep_corrupted), low_frequency_cutoff=args.low_frequency_cutoff)
+                out_segment = out_detector_group.create_dataset(segment_name, data=caster(segment_white_data), compression='gzip', compression_opts=9, shuffle=True)
```

Comment:
The compression option is very compute intensive and often not worth the saved file-size. If the code is parallelized, this might be a strong bottleneck in my experience. Give an option to turn off compression or make compression optional with the default being no compression?